### ISSUE TITLE
Add support for forcing on selinux labels

### ISSUE THREAD

**Issue by cgwalters** [ostreedev/ostree#2804]
Working on https://github.com/containers/bootc/issues/1 I was experimenting with testing a FCOS install from the default GHA Ubuntu VM where SELinux is disabled.  ostree today doesn't work for this, but it should.

Basically, we shouldn't care about the host state, only the target.  If the target system has selinux labels, we should write them.  However, we have to battle libselinux which looks at `/sys/fs/selinux` etc.  Our own sepolicy class also dispatches on that.

Now, when we're operating on a host system that *does* have selinux enabled, I think it's better to use `setfscreatecon()` where we can to ensure atomicity of the security labels.  But if not...I think the patches will look something like this:

```
--- a/src/libostree/ostree-repo-checkout.c
+++ b/src/libostree/ostree-repo-checkout.c
@@ -210,7 +210,7 @@ create_file_copy_from_input_at (OstreeRepo     *repo,
   g_autoptr(GVariant) modified_xattrs = NULL;
 
   /* If we're doing SELinux labeling, prepare it */
-  if (sepolicy_enabled)
+  if (sepolicy_enabled && options->sepolicy->host_enabled())
     {
       /* If doing sepolicy path-based labeling, we don't want to set the
        * security.selinux attr via the generic xattr paths in either the symlink
```

Basically, if the host doesn't have selinux enabled, we just write the `security.selinux` xattr the same as all other xattrs (which works fine with selinux disabled on the host).

**Comment by cgwalters** [comment]
Some more on this in e.g. https://github.com/containers/bootc/pull/397 - we have a warning test there and it reveals tons of missing things for this:

- repo object directories `ostree/repo/objects/a1`
- All repository metadata objects `.dirtree`, `.dirmeta`, `.commit` etc.
- Everything in `ostree/repo/refs`
- Everything under `ostree/deploy`
- Very ironically, the `ostree/deploy/default/var/.ostree-selabeled` file
